modules/interactive_music/editor/audio_stream_interactive_editor_plugin.cpp

```diff
@@ -0,0 +1,413 @@
+		fade_mode->clear();
+		fade_mode->add_icon_item(get_editor_theme_icon(SNAME("FadeDisabled")), TTR("Disabled"), AudioStreamInteractive::FADE_DISABLED);
+		fade_mode->add_icon_item(get_editor_theme_icon(SNAME("FadeIn")), TTR("Fade-In"), AudioStreamInteractive::FADE_IN);
+		fade_mode->add_icon_item(get_editor_theme_icon(SNAME("FadeOut")), TTR("Fade-Out"), AudioStreamInteractive::FADE_OUT);
+		fade_mode->add_icon_item(get_editor_theme_icon(SNAME("FadeCross")), TTR("Cross-Fade"), AudioStreamInteractive::FADE_CROSS);
+		fade_mode->add_icon_item(get_editor_theme_icon(SNAME("AutoPlay")), TTR("Automatic"), AudioStreamInteractive::FADE_AUTOMATIC);
+	}
+}
+
+void AudioStreamInteractiveTransitionEditor::_bind_methods() {
+	ClassDB::bind_method("_update_transitions", &AudioStreamInteractiveTransitionEditor::_update_transitions);
+}
+
+void AudioStreamInteractiveTransitionEditor::_edited() {
+	if (updating) {
+		return;
+	}
+
+	bool enabled = transition_enabled->is_pressed();
+	AudioStreamInteractive::TransitionFromTime from = AudioStreamInteractive::TransitionFromTime(transition_from->get_selected());
+	AudioStreamInteractive::TransitionToTime to = AudioStreamInteractive::TransitionToTime(transition_to->get_selected());
+	AudioStreamInteractive::FadeMode fade = AudioStreamInteractive::FadeMode(fade_mode->get_selected());
+	float beats = fade_beats->get_value();
+	bool use_filler = filler_clip->get_selected() > 0;
+	int filler = use_filler ? filler_clip->get_selected() - 1 : 0;
+	bool hold = hold_previous->is_pressed();
+
+	EditorUndoRedoManager::get_singleton()->create_action("Edit Transitions");
+	for (int i = 0; i < selected.size(); i++) {
+		if (!enabled) {
+			if (audio_stream_interactive->has_transition(selected[i].x, selected[i].y)) {
+				EditorUndoRedoManager::get_singleton()->add_do_method(audio_stream_interactive, "erase_transition", selected[i].x, selected[i].y);
+			}
+		} else {
+			EditorUndoRedoManager::get_singleton()->add_do_method(audio_stream_interactive, "add_transition", selected[i].x, selected[i].y, from, to, fade, beats, use_filler, filler, hold);
+		}
+	}
+	EditorUndoRedoManager::get_singleton()->add_undo_property(audio_stream_interactive, "_transitions", audio_stream_interactive->get("_transitions"));
+	EditorUndoRedoManager::get_singleton()->add_do_method(this, "_update_transitions");
+	EditorUndoRedoManager::get_singleton()->add_undo_method(this, "_update_transitions");
+	EditorUndoRedoManager::get_singleton()->commit_action();
+}
+
+void AudioStreamInteractiveTransitionEditor::_update_selection() {
+	updating_selection = false;
+	int clip_count = audio_stream_interactive->get_clip_count();
+	selected.clear();
+	Vector2i editing;
+	int editing_order = -1;
+	for (int i = 0; i <= clip_count; i++) {
+		for (int j = 0; j <= clip_count; j++) {
+			if (rows[i]->is_selected(j)) {
+				Vector2i meta = rows[i]->get_metadata(j);
+				if (selection_order.has(meta)) {
+					int order = selection_order[meta];
```

Comment:
To be honest, I prefer not do this unless its meant to be high perf code, the code just looks worse for no real reason.

Reply:
Well it originated from this discussion: https://github.com/godotengine/godot/pull/76748#issuecomment-1536292888
I'm trying to use it since then, even if it doesn't really matter. There's also `getptr()`, which looks a bit better.